Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/detail/AgendaDetailFragment.kt

```diff
@@ -246,6 +247,21 @@ class AgendaDetailFragment : Fragment() {
                 scheduleDetail.listRow.talkDescription.getHtmlFormattedSpanned()
 
         tv_agenda_detail_description.movementMethod = LinkMovementMethod.getInstance()
+        tv_agenda_detail_shareText.text = resources.getString(R.string.sharing_twitter_title)
+        imgv_twitter.setOnClickListener {
+            val twitter = viewModel.getSpeaker(scheduleDetail.listRow.primarySpeakerName)?.socialProfiles?.get("twitter")
+            var twitterVal = ""
+            if(twitter.isNullOrEmpty())
+                twitterVal = scheduleDetail.listRow.primarySpeakerName
+            else
+                twitterVal = "@"+twitter
+
+
+            val tweetUrl = "https://twitter.com/intent/tweet?text=I really enjoyed this %23droidconbos talk \"" + scheduleDetail.listRow.talkTitle + "\" by " + twitterVal
```

Comment:
we might be able to make this formatting string a bit more readable by using a string template.  maybe something like:

```suggestion
            val tweetUrl = "https://twitter.com/intent/tweet?text=I really enjoyed this %23droidconbos talk \" ${scheduleDetail.listRow.talkTitle}"\  by $twitterVal"
```